File: problem-solving/algorithms/implementation/utopian-tree/utopian-tree.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the utopianTree function below.
def utopianTree(n):
    if n == 0:
        return 1
    elif n % 2 == 0:
        return utopianTree(n-1)+1
    else:
        return utopianTree(n-1)*2

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input())

    for t_itr in range(t):
        n = int(input())

        result = utopianTree(n)

        fptr.write(str(result) + '\n')

    fptr.close()

File: problem-solving/algorithms/implementation/utopian-tree/utopian-tree.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {

    // Complete the utopianTree function below.
    static int utopianTree(int item) {
        int defaultHeight = 1;
        if (item == 0)
        {
            return defaultHeight;
        }
        else
        {
            bool doubled = false;
            for (int i = 1; i <= item; i++)
            {
                if (!doubled)
                {
                    defaultHeight = defaultHeight * 2;
                    doubled = true;
                }
                else
                {
                    defaultHeight = defaultHeight + 1;
                    doubled = false;
                }
            }

            return defaultHeight;
        }

    }

    static void Main(string[] args) {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int t = Convert.ToInt32(Console.ReadLine());

        for (int tItr = 0; tItr < t; tItr++) {
            int n = Convert.ToInt32(Console.ReadLine());

            int result = utopianTree(n);

            textWriter.WriteLine(result);
        }

        textWriter.Flush();
        textWriter.Close();
    }
}
